audio/src/fetch/receive.rs

```diff
@@ -238,24 +238,24 @@ impl AudioFileFetch {
             // download data from after the current read position first
             let mut tail_end = RangeSet::new();
             let read_position = self.shared.read_position.load(atomic::Ordering::Relaxed);
-            tail_end.add_range(&Range::new(
+            tail_end.add_range(Range::new(
                 read_position,
                 self.shared.file_size - read_position,
             ));
             let tail_end = tail_end.intersection(&missing_data);
 
             if !tail_end.is_empty() {
-                let range = tail_end.get_range(0);
-                let offset = range.start;
-                let length = min(range.length, bytes_to_go);
+                let range = tail_end[0];
```

Comment:
TBH I liked the old version more. I think it's not really obvious what kind of things are indexed, if you understand what I mean.

Reply:
You mean the explicitness of `get_range` as opposed to indexing the `RangeSet`?